shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -0,0 +1,425 @@
+ *  - {@link Activity#onNewIntent(Intent)} ()}
+ *  - {@link Activity#onUserLeaveHint()}
+ *
+ * Additionally, when starting an {@code Activity} for a result from this {@code Fragment}, be sure
+ * to invoke {@link Fragment#startActivityForResult(Intent, int)} rather than
+ * {@link Activity#startActivityForResult(Intent, int)}. If the {@code Activity} version of the
+ * method is invoked then this {@code Fragment} will never receive its
+ * {@link Fragment#onActivityResult(int, int, Intent)} callback.
+ *
+ * If convenient, consider using a {@link FlutterActivity} instead of a {@code FlutterFragment} to
+ * avoid the work of forwarding calls.
+ *
+ * If Flutter is needed in a location that can only use a {@code View}, consider using a
+ * {@link io.flutter.view.FlutterView}. Using a {@link io.flutter.view.FlutterView} requires
+ * forwarding some calls from an {@code Activity}, as well as forwarding lifecycle calls from
+ * an {@code Activity} or a {@code Fragment}.
+ *
+ * @see FlutterActivity
+ *
+ * @see io.flutter.view.FlutterView
+ */
+@SuppressWarnings("WeakerAccess")
+@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1)
+public class FlutterFragment extends Fragment implements PluginRegistry {
+  private static final String TAG = "FlutterFragment";
+
+  private static final String ARG_IS_SPLASH_SCREEN_DESIRED = "show_splash_screen";
+  private static final String ARG_INITIAL_ROUTE = "initial_route";
+  private static final String ARG_APP_BUNDLE_PATH = "app_bundle_path";
+
+  private static final WindowManager.LayoutParams MATCH_PARENT =
+      new WindowManager.LayoutParams(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT);
+
+  public static FlutterFragment newInstance(boolean isSplashScreenDesired,
+                                            @Nullable String initialRoute,
+                                            @NonNull String appBundlePath) {
+    FlutterFragment frag = new FlutterFragment();
+
+    Bundle args = new Bundle();
+    args.putBoolean(ARG_IS_SPLASH_SCREEN_DESIRED, isSplashScreenDesired);
+    args.putString(ARG_INITIAL_ROUTE, initialRoute);
+    args.putString(ARG_APP_BUNDLE_PATH, appBundlePath);
+    frag.setArguments(args);
+
+    return frag;
+  }
+
+  private FrameLayout container;
+  private FlutterView flutterView;
+  private View launchView;
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
+    createLayout();
+    // TODO: Should we start running the FlutterView here or when attached? It was being done here
+    //       in the Activity, but maybe that was a problem to begin with?
+    doInitialFlutterViewRun();
+
+    return container;
+  }
+
+  @Override
+  public void onStart() {
+    super.onStart();
+    flutterView.onStart();
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    // TODO: should flutterView have an onResume() method?
+  }
+
+  public void onPostResume() {
```

Comment:
Should the fragment be listening to activity lifecycle instead of relying on this being called directly?

Reply:
I don't think so.  I mentioned in the doc related to this work that I'm setting up unidirectional information flow from Activity -> Fragment -> View.  If a specific reason arises to reconsider that restriction then we can adjust as needed.